### PR TITLE
Fix: workflow skip executing all steps occasionally

### PR DESCRIPTION
**Pull request by leejanee**
Signed-off-by: Jian.Li <lj176172@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by yangsoon — APPROVED** [review]
lgtm

File: pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go [hunk 1]
```diff
@@ -375,7 +377,12 @@ func (r *Reconciler) patchStatus(ctx context.Context, app *v1beta1.Application,
 func (r *Reconciler) updateStatus(ctx context.Context, app *v1beta1.Application, phase common.ApplicationPhase) error {
 	app.Status.Phase = phase
 	updateObservedGeneration(app)
-	return r.Status().Update(ctx, app)
+
+	obj, err := app.Unstructured()
+	if err != nil {
+		return err
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go:383]: let's make it configurable？

File: cmd/core/main.go [hunk 2]
```diff
@@ -131,6 +131,7 @@ func main() {
 	flag.DurationVar(&retryPeriod, "leader-election-retry-period", 2*time.Second,
 		"The duration the LeaderElector clients should wait between tries of actions")
 	flag.BoolVar(&enableClusterGateway, "enable-cluster-gateway", false, "Enable cluster-gateway to use multicluster, disabled by default.")
+	flag.BoolVar(&controllerArgs.Enable3025, "enable-3025", false, "prune")
```
Comment (wonderflow) [inline cmd/core/main.go:134]:
```suggestion
	flag.BoolVar(&controllerArgs.Enable3025, "compatible-status-update", false, "prune")
```

File: pkg/controller/core.oam.dev/v1alpha2/application/suite_test.go [hunk 3]
```diff
@@ -137,13 +137,14 @@ var _ = BeforeSuite(func(done Done) {
...
-		Recorder:         event.NewAPIRecorder(recorder),
-		appRevisionLimit: appRevisionLimit,
+		Client:   k8sClient,
+		Scheme:   testScheme,
+		dm:       dm,
+		pd:       pd,
+		Recorder: event.NewAPIRecorder(recorder),
 	}
+
+	reconciler.concurrentReconciles = appRevisionLimit
```
Comment (wonderflow) [inline pkg/controller/core.oam.dev/v1alpha2/application/suite_test.go:147]: typo？

**Review by hongchaodeng — APPROVED** [review]

**Review by wonderflow — APPROVED** [review]

**Pull request merged**